script/vsts/platforms/linux.yml

```diff
@@ -24,10 +24,10 @@ jobs:
 
       - task: NodeTool@0
         inputs:
-          versionSpec: 12.13.1
-        displayName: Install Node.js 12.13.1
+          versionSpec: 12.4.0
+        displayName: Install Node.js 12.4.0
 
-      - script: npm install --global npm@6.12.1
+      - script: npm install --global npm@6.9.0
```

Comment:
Hi all.

In my opinion it could be good to have _newer_ `npm` in CI, rather than pinning to `npm@6.9.0`.

The system- wide npm is only used to install `script/package.json`, after which point the `script/node_modules/.bin/npm` is used to install `apm`, and `apm` is used to install the main packages.

System-wide npm is also used to install CI dependencies (`script/vsts/node_modules` and things like that). Again, here too it would be nice to have recent npm for speed, up-to-date dependency tree algorithm, and various bug-fixes.

In my view, just having an up-to-date npm is the priority. As various small bugs have been fixed over time. Some of the old bugs are inconvenient for things we are trying to do at our fork, so we would possibly patch the CI configs at our fork to have recent `npm`. Also: old `npm` (older than 6.12.0) doesn't work with Python 3. So at least that version would be nice. (Apparently all the CI environments have Python 2, so it's not a huge deal.)

Reply:
Similar thought about Node 12. Newer is mostly better, although 12.16.x is the newest one can go in 12.x without running into a problem: https://github.com/atom/atom/issues/21091

Reply:
I would say let's merge this as it is, and implement your suggestions in separate pull requests. This is well tested by us as of now.